### ISSUE TITLE
Spike: Find a shorthand for filters and mappers

### ISSUE THREAD

**Issue by JanLenoch** [Kentico/xperience-training-13#35]
### Expected result

The future document repository class will use Func<> delegates for ad-hoc filters and mappers. However, writing `Func<CompanyService, CompanyServiceDto, CompanyServiceDto> Mapper` is cumbersome. Find a way to use a shorthand.

### Additional context

Add any other context or guidelines here.

### Resources

* https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/delegates/

### Issue expands upon

This issue expands upon another issue(s): #26 

### Blockers

This issue is blocked by the following issue(s): #`issue number`
